gtsam/nonlinear/LevenbergMarquardtOptimizer.cpp

```diff
@@ -158,10 +158,13 @@ bool LevenbergMarquardtOptimizer::tryLambda(const GaussianFactorGraph& linear,
     if (params_.verbosityLM >= LevenbergMarquardtParams::TRYDELTA)
       delta.print("delta");
 
-    // cost change in the linearized system (old - new)
+    // Compute the old linearized error as it is not the same
+    // as the nonlinear error when robust noise models are used.
+    double oldLinearizedError = linear.error(VectorValues::Zero(delta));
```

Comment:
I'm OK with merging this if it fixes the robust noise issues now, but we should investigate computational cost. Also, can we cache this Zero call somewhere?

Reply:
Yeah this is indeed a computation burden :( I believe this will stay the same over the entire execution of LM so we can definitely cache this (a const variable outside of the loop would suffice).